include/kllvm/util/temporary_file.h

```diff
@@ -0,0 +1,50 @@
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <unistd.h>
+
+typedef std::unique_ptr<FILE, decltype(&fclose)> temp_c_file_type;
+
+class temporary_file {
+private:
+  int temp_fd;
+  std::string temp_filename;
+  temp_c_file_type temp_c_file;
+  std::optional<std::ofstream> temp_cpp_file;
+
+public:
+  temporary_file(std::string template_name)
+      : temp_c_file(nullptr, &fclose) {
+    temp_fd = mkstemp(template_name.data());
+    temp_filename = template_name;
+
+    if (temp_fd == -1) {
+      std::runtime_error("Could not create temporary file!");
+    }
+  }
+
+  ~temporary_file() {
+    close(temp_fd);
+    remove(temp_filename.data());
```

Comment:
This can be a call to `std::filesystem::remove`, I think: https://en.cppreference.com/w/cpp/filesystem/remove

Reply:
No problem - nice digging